rfc/rfc006-distributed-registry.md

```diff
@@ -323,18 +329,62 @@ The care organisation refers to it:
   "id": "did:nuts:abc",
   "service": [
     {
-      "id": "did:nuts:abc#NutsCompoundService-1",
+      "id": "did:nuts:abc#F1Dsgwngfdg3SH6TpDv0Ta1aOE",
       "type": "NutsCompoundService",
       "serviceEndpoint": {
-        "oauth": "did:nuts:123#NutsOAuth",
-        "fhirEndpoint": "did:nuts:123#NutsFHIR"
+        "oauth": "did:nuts:123?type=oauth",
+        "fhirEndpoint": "did:nuts:123?type=fhir"
       }
     }
   ]
 }
 ```
 
-## 5. Supported Cryptographic Algorithms and Key Types
+## 5. Conflict resolution
+
+§3.4 of [RFC004](rfc004-verifiable-transactional-graph.md) requires each payload type to describe how conflicts are resolved when parallel transactions are encountered.
+A DID Document is not a CRDT, but its contents can be. The paragraphs below describe how each of the elements should be treated in case of a conflict.
+First we describe the mechanism of detecting and resolving conflicts. DID Documents do not refer to their previous version in any way. 
+The transactions described in [RFC004](rfc004-verifiable-transactional-graph.md) do refer to previous transactions. This system can be used to detect conflicts and to resolve them.
+For DID Documents additional transactional requirements are added:
+
+* Updates to DID Documents MUST refer in their transaction to the transaction of the previous version. (In addition to the HEADs)
+* In case of a conflict, the conflict can be resolved with an additional transaction referring to the all the conflicting transactions.
+
+The algorithm for conflict detection can be referred from these requirements:
+
+- A DID Document update transaction is received.
+- The current version is resolved together with the metadata (the metadata holds the originating transaction(s)).
+- If the transaction refers to all transactions present in the metadata, the update is valid and the DID Document is replaced.
+- If the transaction does not refer to all transactions, it is a conflict and the rules from the paragraphs below are applied. 
+
+### 5.1 @context
+
+The `@context` is determined by the protocol version and should not change.
+
+### 5.2 id
+
+The `id` identifies the DID Document and by definition will be equal for updates.
+
+### 5.3 controller
+
+Controller entries are references to other DID Documents. If the list of controllers differs, the resulting list must be all unique elements from both documents combined.
+
+### 5.4 verification methods
+
+Verification methods are defined by their public-key and thus their contents. This makes each method immutable. When the list of methods differs between documents,
```

Comment:
public key


this also means we can only support immutable verification method types

Reply:
which they already are.